components/datetime/src/time_zone.rs

```diff
@@ -888,7 +888,7 @@ pub(super) trait FormatTimeZoneWithFallback {
             self.format_gmt_offset(sink, gmt_offset, data_payloads)?
         } else {
             sink.with_part(Part::ERROR, |sink| {
-                sink.write_str(&zone_formats.gmt_offset_fallback)
+                sink.write_str(&zone_formats.gmt_format.replace("{0}", "+?"))
```

Comment:
I am turning a blind eye to the allocation here since there is another issue where we're supposed to clean up the data struct and make it use `icu_pattern`.